data_extrator.py

```diff
@@ -22,14 +18,26 @@ def exponential_filter(time_step):
         return filtered_data[-parameters_per_slice:]
 
 
-    parameters = np.zeros((slice_count, parameters_per_slice))
-    for i in range(slice_count):
-        # Calculate last point of slice
-        time_step = time_step_0 - i*slice_shift
-        # Calculate last 6 points of filtered values
-        parameters[i] = exponential_filter(time_step)
+    def exponential_filter_parameters(data, time_step_0, slice_count=5, slice_overlap=50, alpha_value = 0.8, slice_size = 100, parameters_per_slice = 6, plot_results = False):
+        # Check if enough data points
+        slice_shift = slice_size - slice_overlap
+        if time_step_0 - (slice_count-1) * slice_shift - slice_size < 0:
+            raise ValueError("Dataset is too small to extract parameters.")
+        if time_step_0 > len(data) - 1:
```

Comment:
To chyba jest zbyt słaby warunek. time_step_0 musi być mniejszy niż rozmiar dataset'u minus rozmiar przesuwanego okna (jak w powyższym komentarzu)